File: SoftUni - C# OOP/Interfaces and Abstraction/Interfaces and Abstraction - Exercise/03. Telephony/Models/Smartphone.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace _03._Telephony.Models.Interfaces
{
    public class Smartphone : ICallable, IBrawseble
    {
        public string BrawsURL(string url)
        {
            return $"Browsing: {url}!";
        }

        public string Call(string number)
        {
            return $"Calling... {number}";
        }
    }
}

File: SoftUni - Java OOP/Interfaces and Abstraction/Interfaces and Abstraction - Exercises/_05_Telephony/Smartphone.java
package _05_Telephony;
import java.util.List;

public class Smartphone implements Callable, Browsable{

    private List<String> numbers;
    private List<String> urls;

    public Smartphone(List<String> numbers, List<String> urls) {
        this.numbers = numbers;
        this.urls = urls;
    }

    @Override
    public String browse() {
        StringBuilder sb = new StringBuilder();
        for (String url : urls) {

            if (!url.matches(".*[0-9].*")) {
                sb.append("Browsing: ").append(url).append("!");
            }

            else {
                sb.append("Invalid URL!");
            }
            sb.append(System.lineSeparator());
        }
        return sb.toString().trim();
    }

    @Override
    public String call() {
        StringBuilder sb = new StringBuilder();
        for (String currentNumber : numbers) {

            if (currentNumber.matches("[0-9]+")) {
                sb.append("Calling... ").append(currentNumber);
            }

            else {
                sb.append("Invalid number!");
            }
            sb.append(System.lineSeparator());
        }
        return sb.toString().trim();
    }
}
